web/src/main/webapp/WEB-INF/jsp/managestudy/findSubjects.jsp

```diff
@@ -51,8 +57,113 @@
+                                                            <fmt:message key='pidv_err' bundle='${resword}'/> \
+                                                        </div> \
+                                                        <div id='pidv-match' style='display: none; color: #0cb924;' class='alert small'> \
+                                                            <fmt:message key='pidv_match' bundle='${resword}'/> \
+                                                        </div> \
+                                                    </div>\
+                                                </td> \
+                                            </tr> \
+                                        </table> \
+                                    </td> \
+                                </tr> \
+                                <tr> \
+                                    <td colspan='2' style='text-align: center;'> \
+                                            <input type='button' value='Cancel' onclick='clearPIDVerificationForm()'/> \
+                                            &nbsp; \
+                                            <input type='button' value='Checking' onclick='validatePIDVerificationForm()'/> \
+                                    </td> \
+                                </tr> \
+                            </table>"
+                }],
+                template: "<div class='popover tour'> \
+                            <div class='arrow'></div> \
+                            <h3 class='popover-title'></h3> \
+                            <div class='popover-content'></div> \
+                            <nav class='popover-navigation' style='display:none;'> \
+                                <button class='btn btn-default' data-role='prev'>« Prev</button> \
+                                <span data-role='separator'>|</span> \
+                                <button class='btn btn-default' data-role='next'>Next »</button> \
+                                <button class='btn btn-default' data-role='end'>End tour</button> \
+                            </nav> \
+                            </div>",
+                onEnd: function(t) {
+                    // need to found the correct way to handle this default setting
+                    jQuery("#pid-" + currentPID).css({'display':'block'});
+                },
+                onShown: function(t) {
+                    // disable right click
+                    jQuery("#step-0").find('#retype_pid').on('contextmenu',function(){
+                        return false;
+                    });
+                    // disable cut copy paste
+                    jQuery("#step-0").find('#retype_pid').bind('cut copy paste', function (e) {
+                        e.preventDefault();
+                    });
+                }
+            });
+
+            tourElement.init();
+            tourElement.start(true);
+        });
+
+
     });
```

Comment:
please move the code above and the functions below into a separate javascript file that can be included based on whether the datainfo property. The idea here is to separate the custom html/javascript from the main codebase and include it conditionally.